spec/models/edited_location_spec.rb

```diff
@@ -25,28 +25,100 @@
     end
   end
 
-  describe '#initialize' do
-    it 'delegates to the newest location in the array' do
-      result = subject.new([location_2, location_1])
-      expect(result.__getobj__).to eq(location_2)
-    end
+  %i(uid title booking_location).each do |field|
+    describe "##{field}" do
+      it 'delegates to the newest location in the array based on version number' do
+        result = subject.new([location_2, location_1])
+        expect(result.public_send(field)).to eq(location_2.public_send(field))
 
-    context 'when no previous location can be found' do
-      it 'gets all the edited location fields for each location' do
-        expect(EditedLocationField).to receive(:all).with(location_1, nil).and_call_original
-        subject.new([location_1])
+        result = subject.new([location_1, location_2])
+        expect(result.public_send(field)).to eq(location_2.public_send(field))
       end
     end
+  end
+
+  describe '.edits' do
+    subject { described_class.new(versions) }
+
+    let(:new_location) { location(version: 2) }
+    let(:old_location) { location(version: 1) }
+    let(:versions) { [new_location, old_location] }
+
+    let(:edit) { subject.edits.values.flatten.first }
 
-    context 'when a previous location can be found' do
+    it 'returns an edit record for each field that has changed' do
+      new_location.title = 'Beta'
+
+      expect(subject.edits.count).to eq(1)
+    end
+
+    context 'returned object is correctly populated can on edit details' do
```

Comment:
This doesn't make any sense. Do you mean `returned object is correctly populate upon edit` or something similar?